### PR TITLE
JS: add support for util.promisify with child_process calls

### PR DESCRIPTION
**Pull request by erik-krogh**
Adds support for the sink in CVE-2019-10776

A [quick smoke-test](https://git.semmle.com/erik/dist-compare-reports/tree/profiling-js-max.northeurope.cloudapp.azure.com_1586966334818) reveals decent performance, and what looks like a new TP.

### CODE REVIEW

**Comment by erik-krogh** [comment]
> It is probably a pipe dream to have promisify work for all callback based models automatically. 

I think we can do it if we treat it similarly to reflective calls (and create a new `CallNodeDef` in `DataFlow.qll`).  
But I haven't found a nice way to do it without lots of non-monotonic recursion.  
So probably best not to go that route.

**Comment by erik-krogh** [comment]
> Perhaps we should just tag the new predicate with EXPERIMENTAL, and perhaps do a large study later for how valuable the extra call node could be. 

Or make it private? 

> perhaps do a large study later for how valuable the extra call node could be.
I'll look into that after this is merged.

**Comment by esbena** [comment]
> Or make it private?

Yes. I did not realise that both clients of the predicate were in the same file.

File: change-notes/1.24/analysis-javascript.md [hunk 1]
```diff
@@ -87,6 +87,8 @@
 | Identical operands (`js/redundant-operation`) | Fewer results | This query now recognizes cases where the operands change a value using ++/-- expressions. |
 | Superfluous trailing arguments (`js/superfluous-trailing-arguments`) | Fewer results | This query now recognizes cases where a function uses the `Function.arguments` value to process a variable number of parameters. |
 | Incomplete URL scheme check (`js/incomplete-url-scheme-check`) | More results | This query now recognizes more variations of URL scheme checks. |
+| Uncontrolled data used in path expression (`js/path-injection`) | More results | This query now recognizes more file system calls. |
+| Uncontrolled command line (`js/command-line-injection`) | More results | This query now recognizes more command execution calls. |
```
Comment (esbena) [inline change-notes/1.24/analysis-javascript.md:91]: `additional` seems to be the preferred over `more` (also fixes an older PR).
```suggestion
| Superfluous trailing arguments (`js/superfluous-trailing-arguments`) | Fewer results | This query now recognizes cases where a function uses the `Function.arguments` value to process a variable number of parameters. |
| Incomplete URL scheme check (`js/incomplete-url-scheme-check`) | More results | This query now recognizes additional variations of URL scheme checks. |
| Uncontrolled data used in path expression (`js/path-injection`) | More results | This query now recognizes additional file system calls. |
| Uncontrolled command line (`js/command-line-injection`) | More results | This query now recognizes additional command execution calls. |
```

**Comment by erik-krogh** [comment]
[A larger evaluation](https://git.semmle.com/erik/dist-compare-reports/tree/profiling-js-max.northeurope.cloudapp.azure.com_1587197230110) looks just fine.

**Pull request merged**